edx/edx-ora2
# [BD-5] [TNL-7274] BB-2504: Add grade status summary page

Add a message on the “Your grade” step explaining how the final grade is calculated/determined.

**JIRA tickets**: 
* https://openedx.atlassian.net/browse/TNL-7274
* https://openedx.atlassian.net/browse/OSPR-4663

**Dependencies**: None

**Screenshots**: 
**[Flow: Self Assessment Only]**
![image](https://user-images.githubusercontent.com/3592998/86166658-91f71400-bb2e-11ea-9ab8-c4aa06bd49c5.png)

**[Flow: Peer Assessment Only]**
![image](https://user-images.githubusercontent.com/3592998/86166736-b0f5a600-bb2e-11ea-9d9a-8b1c704e9f46.png)

**[Flow: Staff Assessment Only]**
![image](https://user-images.githubusercontent.com/3592998/86166765-be129500-bb2e-11ea-9bfb-9b6465067918.png)

**[Flow: If staff assessment is one of the assessment steps and all the assessments are not complete yet]**
![image](https://user-images.githubusercontent.com/3592998/86166998-2c575780-bb2f-11ea-95da-35c7c76fec9f.png)

**[Flow: Peer with incomplete peer reviews]**
![image](https://user-images.githubusercontent.com/3592998/86166933-10ec4c80-bb2f-11ea-80f2-55d70c02e7f9.png)

**[Flow: Self Assessment Only but when it is not done yet]**
![image](https://user-images.githubusercontent.com/3592998/86167065-501a9d80-bb2f-11ea-8d2b-bb6265146654.png)

**Merge deadline**: None

**Testing instructions**:

1. Create a sample course.
2. Create ORA (Open Response Assessment) one for each assessment type e.g **self**, **peer**, **staff**, **self** + **peer**, ....., etc.
3. Create student and enroll them into the created course.
4. Get them to attempt the ORA assessments and get them to grading page.
5. Check the output HTML and see whether the grade explanation message match the following criteria.
    ### If all the assessments are complete

    * If there is a **Staff Assessment** then it should say 
        >You have successfully completed this problem and received a {earned_points}/{total_points}. The grade for this problem is determined by your Staff Grade.
    
        no matter whether there is also a **Peer** or **Self Assessment** or not as Staff Assessments overrides every other type of assessment.
    * If there is only a single type of assessment be it **Staff, Self or Peer**, it should output the following message corresponding to each assessment type
        * **Staff**: You have successfully completed this problem and received a {earned_points}/{total_points}. The grade for this problem is determined by your Staff Grade.
        * **Peer**: You have successfully completed this problem and received a {earned_points}/{total_points}. The grade for this problem is determined by the median score of your Peer Assessments.
        * **Self**: You have successfully completed this problem and received a {earned_points}/{total_points}. The grade for this problem is determined by your Self Assessment.
    * If an assignment have both **Self** and Peer **Assessment** than the output message would depend upon the `Grade Assessment Priority`. By default it is as follows
        1. Staff Assessment
        2. Peer Assessment
        3. Self Assessment

        So, in case of default assessment priority the message would be 
        >You have successfully completed this problem and received a {earned_points}/{total_points}. The grade for this problem is determined by the median score of you Peer Assessments.
        
        but if the **Self Assessment** have higher priority than **Peer Assessment** than the message would be
        >You have successfully completed this problem and received a {earned_points}/{total_points}. The grade for this problem is determined by your Self Assessment.

    ### If some assessment is incomplete

    * No message should contain the sentence 
        >You have successfully completed this problem and received a {earned_points}/{total_points}.
    * The second sentence of messages would follow the above rules.
    * First sentence of all flows would be empty except for one special case and that is **Peer Assessment** only. It should have the following first sentence
        >You have not yet received all necessary peer reviews to determine your final grade.

**Reviewers**
